GoogleDataTransport/ProtoSupport/Protos/client_metrics.proto

```diff
@@ -0,0 +1,79 @@
+syntax = "proto3";
+
+package gdt_client_metrics;
+
+// Top level metrics for all client analytics metrics.
+// These metrics should be sent as a part of every request that is uploaded to
+// FireLog server. In more detail, an additional LogRequest should be added to
+// the BatchedLogRequest, where the LogSource of the LogRequest should be
+// GDT_CLIENT_METRICS and the LogRequest should have a single LogEvent whose
+// payload is a GoogleDataTransportClientMetrics message.
+//
+// Original proto: google3/firebase/firelog/proto/gdt_client_analytics.proto
+// See go/firelog-client-analytics for more details.
+message ClientMetrics {
+  // The window of time over which the metrics are evaluated.
+  TimeWindow window = 1;
+
+  repeated LogSourceMetrics log_source_metrics = 2;
+
+  GlobalMetrics global_metrics = 3;
+
+  // The bundle ID on Apple platforms (e.g., iOS) or the package name on Android
+  string app_namespace = 4;
+}
+
+// Represents an arbitrary window of time.
+message TimeWindow {
+  // The time that the window first starts.
+  // start_ms is the number of milliseconds since the UNIX epoch
+  // (January 1, 1970 00:00:00 UTC)
+  int64 start_ms = 1;
+
+  // The time that the window ends.
+  // end_ms is the number of milliseconds since the UNIX epoch
+  // (January 1, 1970 00:00:00 UTC)
+  int64 end_ms = 2;
+}
+
+// Metrics per app, not per log source
+message GlobalMetrics {
+  StorageMetrics storage_metrics = 1;
+}
+
+message StorageMetrics {
+  // The number of bytes of storage the event cache was consuming on the client
+  // at the time the request was sent.
+  int64 current_cache_size_bytes = 1;
+
+  // The maximum number of bytes to which the event cache is allowed to grow.
+  int64 max_cache_size_bytes = 2;
+}
+
+// Metrics per log source.
+message LogSourceMetrics {
+  // A LogSource uniquely identifies a logging configuration. log_source should
+  // contains a string value of the LogSource from
+  // google3/wireless/android/play/playlog/proto/clientanalytics.proto
+  string log_source = 1;
+
+  repeated LogEventDropped log_event_dropped = 2;
```

Comment:
Based on DD I'm referencing, is this supposed to have `num_data_events` and `num_telemetry_events` fields?

Reply:
same as [above](https://github.com/google/GoogleDataTransport/pull/40/files#r717703604)